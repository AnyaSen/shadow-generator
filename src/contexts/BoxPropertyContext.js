import React, { useState, createContext } from "react";

export const BoxPropertyContext = createContext();

export const BoxPropertyProvider = ({ children }) => {
  const [heightValue, setHeightValue] = useState(350);
  const [widthValue, setWidthValue] = useState(350);
  const [radiusValue, setRadiusValue] = useState(16);
  const [borderValue, setBorderValue] = useState(0);

  const ChangeHeightValue = (event, newValue) => {
    setHeightValue(newValue);
  };

  const ChangeWidthValue = (event, newValue) => {
    setWidthValue(newValue);
  };

  const ChangeBorderValue = (event, newValue) => {
    setBorderValue(newValue);
  };

  const ChangeRadiusValue = (event, newValue) => {
    setRadiusValue(newValue);
  };

  return (
    <BoxPropertyContext.Provider
      value={{
        heightValue,
        ChangeHeightValue,
        widthValue,
        ChangeWidthValue,
        borderValue,
        ChangeBorderValue,
        radiusValue,
        ChangeRadiusValue
      }}
    >
      {children}
    </BoxPropertyContext.Provider>
  );
};
