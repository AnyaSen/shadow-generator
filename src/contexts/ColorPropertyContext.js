import React, { useState, createContext } from "react";

export const ColorPropertyContext = createContext();

export const ColorPropertyProvider = ({ children }) => {
  const [boxColor, setBoxColor] = useState("#f3f3f3");
  const [shadowColor, setShadowColor] = useState("#999999");
  const [bgColor, setBgColor] = useState("#fff");
  const [borderColor, setBorderColor] = useState("#808080");

  const ChangeBoxColor = newColor => {
    setBoxColor(newColor.hex);
  };
  const ChangeShadowColor = newColor => {
    setShadowColor(newColor.hex);
  };
  const ChangeBgColor = newColor => {
    setBgColor(newColor.hex);
  };
  const ChangeBorderColor = newColor => {
    setBorderColor(newColor.hex);
  };

  return (
    <ColorPropertyContext.Provider
      value={{
        boxColor,
        ChangeBoxColor,
        ChangeShadowColor,
        shadowColor,
        ChangeBgColor,
        bgColor,
        ChangeBorderColor,
        borderColor
      }}
    >
      {children}
    </ColorPropertyContext.Provider>
  );
};
