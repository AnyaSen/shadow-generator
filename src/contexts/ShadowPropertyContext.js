import React, { useState, createContext } from "react";

export const ShadowPropertyContext = createContext();

export const ShadowPropertyProvider = ({ children }) => {
  const [shiftHorizontically, setShiftHorizontically] = useState(20);
  const [shiftVertically, setShiftVertically] = useState(14);
  const [blurValue, setBlurValue] = useState(10);
  const [opacityValue, setOpacityValue] = useState(12);

  const ChangeShiftHorizontically = (event, newValue) => {
    setShiftHorizontically(newValue);
  };

  const ChangeShiftVertically = (event, newValue) => {
    setShiftVertically(newValue);
  };

  const ChangeBlurValue = (event, newValue) => {
    setBlurValue(newValue);
  };

  const ChangeOpacityValue = (event, newValue) => {
    setOpacityValue(newValue);
  };

  return (
    <ShadowPropertyContext.Provider
      value={{
        shiftHorizontically,
        ChangeShiftHorizontically,
        shiftVertically,
        ChangeShiftVertically,
        blurValue,
        ChangeBlurValue,
        opacityValue,
        ChangeOpacityValue
      }}
    >
      {children}
    </ShadowPropertyContext.Provider>
  );
};
