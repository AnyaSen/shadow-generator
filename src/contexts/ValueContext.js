import React, { useState, createContext } from "react";

export const ValueContext = createContext();

export const ValueProvider = ({ children }) => {
  const [heightValue, setHeightValue] = useState(45);
  const [widthValue, setWidthValue] = useState(45);
  const [radiusValue, setRadiusValue] = useState(1);
  const [shiftHorizontically, setShiftHorizontically] = useState(1.25);
  const [shiftVertically, setShiftVertically] = useState(0.85);
  const [blurValue, setBlurValue] = useState(0.625);
  const [opacityValue, setOpacityValue] = useState(0.75);
  const [borderValue, setBorderValue] = useState(0);
  const [boxColor, setBoxColor] = useState("#f3f3f3");
  const [shadowColor, setShadowColor] = useState("#999999");
  const [bgColor, setBgColor] = useState("#fff");
  const [borderColor, setBorderColor] = useState("#808080");

  const ChangeHeightValue = (event, newValue) => {
    setHeightValue(newValue);
  };
  const ChangeWidthValue = (event, newValue) => {
    setWidthValue(newValue);
  };
  const ChangeBorderValue = (event, newValue) => {
    setBorderValue(newValue);
  };
  const ChangeRadiusValue = (event, newValue) => {
    setRadiusValue(newValue);
  };
  const ChangeShiftHorizontically = (event, newValue) => {
    setShiftHorizontically(newValue);
  };
  const ChangeShiftVertically = (event, newValue) => {
    setShiftVertically(newValue);
  };
  const ChangeBlurValue = (event, newValue) => {
    setBlurValue(newValue);
  };
  const ChangeOpacityValue = (event, newValue) => {
    setOpacityValue(newValue);
  };
  const ChangeBoxColor = newColor => {
    setBoxColor(newColor.hex);
  };
  const ChangeShadowColor = newColor => {
    setShadowColor(newColor.hex);
  };
  const ChangeBgColor = newColor => {
    setBgColor(newColor.hex);
  };
  const ChangeBorderColor = newColor => {
    setBorderColor(newColor.hex);
  };

  return (
    <ValueContext.Provider
      value={{
        heightValue,
        ChangeHeightValue,
        widthValue,
        ChangeWidthValue,
        borderValue,
        ChangeBorderValue,
        radiusValue,
        ChangeRadiusValue,
        shiftHorizontically,
        ChangeShiftHorizontically,
        shiftVertically,
        ChangeShiftVertically,
        blurValue,
        ChangeBlurValue,
        opacityValue,
        ChangeOpacityValue,
        boxColor,
        ChangeBoxColor,
        ChangeShadowColor,
        shadowColor,
        ChangeBgColor,
        bgColor,
        ChangeBorderColor,
        borderColor
      }}
    >
      {children}
    </ValueContext.Provider>
  );
};
