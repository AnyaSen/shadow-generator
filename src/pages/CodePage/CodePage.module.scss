@import "../../assets/scss/variables";
@import "../../assets/scss/breakpoints";

.CodePage {
  display: flex;
  flex-direction: column;
  align-items: center;

  h2 {
    color: $color-dark-grey;
  }

  .CodeContainer {
    border: 0.1rem solid $color-mulberry;
    border-radius: 0.3rem;
    overflow-y: auto;
    width: 50%;
    height: fit-content;
    padding: $space-1;

    @include lg {
      width: 70%;
    }

    @include md {
      width: 90%;
    }

    .code {
      font-size: $font-md;

      @include sm {
        font-size: $font-sm;
      }
    }
  }

  .BtnContainer {
    display: flex;
    justify-content: space-around;
  }
}
