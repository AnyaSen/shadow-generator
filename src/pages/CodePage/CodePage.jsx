import React, { useContext, useState, useRef, useEffect } from "react";
import Styles from "./CodePage.module.scss";

import { Link } from "react-router-dom";

import Layout from "../../components/Layout/Layout";
import Button from "../../components/Buttons/Button/Button";
import CopyCodeToClipboard from "./CopyCodeToClipboard/CopyCodeToClipboard";
import CodeTemplate from "./CodeTemplate/CodeTemplate";

import { BoxPropertyContext } from "../../contexts/BoxPropertyContext";
import { ColorPropertyContext } from "../../contexts/ColorPropertyContext";
import { ShadowPropertyContext } from "../../contexts/ShadowPropertyContext";

export default function CodePage() {
  const { heightValue, widthValue, borderValue, radiusValue } = useContext(
    BoxPropertyContext
  );

  const {
    shiftHorizontically,
    shiftVertically,
    opacityValue,
    blurValue
  } = useContext(ShadowPropertyContext);

  const { boxColor, shadowColor, borderColor, bgColor } = useContext(
    ColorPropertyContext
  );

  const [copyValue, setCopyValue] = useState("");
  const [copied, setCopied] = useState(false);

  const codeElement = useRef(null);

  useEffect(() => {
    setCopyValue(codeElement.current.textContent);
  }, []);

  const changeCopied = () => {
    setCopied(true);
    setTimeout(() => {
      setCopied(false);
    }, 1000);
  };

  const handleClickCopy = () => {
    setCopyValue(codeElement.current.textContent);
  };

  return (
    <Layout codePage>
      <div className={Styles.CodePage} data-testid="codePage">
        <div className={Styles.CodeContainer} ref={codeElement}>
          <CodeTemplate
            heightValue={heightValue}
            widthValue={widthValue}
            borderValue={borderValue}
            radiusValue={radiusValue}
            boxColor={boxColor}
            shadowColor={shadowColor}
            borderColor={borderColor}
            bgColor={bgColor}
            shiftHorizontically={shiftHorizontically}
            shiftVertically={shiftVertically}
            opacityValue={opacityValue}
            blurValue={blurValue}
          />
        </div>

        <div className={Styles.BtnContainer}>
          <CopyCodeToClipboard
            copyValue={copyValue}
            changeCopied={changeCopied}
            onClick={handleClickCopy}
          />

          <Link to="/">
            <Button text="Go back" />
          </Link>
        </div>

        {copied ? <h2>Copied!</h2> : null}
      </div>
    </Layout>
  );
}
