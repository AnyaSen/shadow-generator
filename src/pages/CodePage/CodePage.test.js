import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";

import CodePage from "./CodePage";

import { BoxPropertyProvider } from "../../contexts/BoxPropertyContext";
import { ShadowPropertyProvider } from "../../contexts/ShadowPropertyContext";
import { ColorPropertyProvider } from "../../contexts/ColorPropertyContext";

describe("<CodePage />", () => {
  afterEach(cleanup);

  const createRenderTree = () => (
    <BrowserRouter>
      <BoxPropertyProvider>
        <ShadowPropertyProvider>
          <ColorPropertyProvider>
            <CodePage />
          </ColorPropertyProvider>
        </ShadowPropertyProvider>
      </BoxPropertyProvider>
    </BrowserRouter>
  );

  test("Should return 2 or more children.", () => {
    const tree = createRenderTree();

    const { getByTestId, container } = render(tree);

    const shadowPropsContainer = getByTestId(`codePage`);

    expect(shadowPropsContainer.children.length).toBeGreaterThanOrEqual(2);

    expect(container.firstChild).toMatchSnapshot();
  });
});
