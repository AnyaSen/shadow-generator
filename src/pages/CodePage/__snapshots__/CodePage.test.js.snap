// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodePage /> Should return 2 or more children. 1`] = `
<div
  class="Layout"
  data-testid="Layout"
>
  <div
    class="Header"
  >
    <a
      href="/"
    >
      <h1>
        SHADOW GENERATOR
      </h1>
    </a>
  </div>
  <div
    class="childrenContainer"
  >
    <div
      class="CodePage"
      data-testid="codePage"
    >
      <div
        class="CodeContainer"
      >
        <pre
          class="code"
        >
          .backgroundContainer{ 
          background-color: #fff

          .shadowBox{
            height: 200px;
            width: 200px;
            box-shadow: 20px 14px 10px rgba(153, 153, 153, 12);
            border-radius: 16px;
            background-color: #f3f3f3;
            
}
}
        </pre>
      </div>
      <div
        class="BtnContainer"
      >
        <span>
          <button
            class="Button"
          >
            Copy
          </button>
        </span>
        <a
          href="/"
        >
          <button
            class="Button"
          >
            Go back
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    class="Footer"
  >
    <p>
      Created by
       
      <a
        href="https://portfolio-anna-senchikhina.netlify.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Anna Senchikhina
      </a>
    </p>
  </div>
</div>
`;
